private/buf/buflsp/file.go

```diff
@@ -170,6 +198,99 @@ func (f *file) Update(ctx context.Context, version int32, text string) {
 	f.hasText = true
 }
 
+// FetchSettings refreshes configuration settings for this file.
+//
+// This only needs to happen when the file is open or when the client signals
+// that configuration settings have changed.
+func (f *file) RefreshSettings(ctx context.Context) {
+	settings, err := f.lsp.client.Configuration(ctx, &protocol.ConfigurationParams{
+		Items: []protocol.ConfigurationItem{
+			{ScopeURI: f.uri, Section: ConfigBreakingStrategy},
+			{ScopeURI: f.uri, Section: ConfigBreakingGitRef},
+		},
+	})
+	if err != nil {
+		// We can throw the error away, since the handler logs it for us.
+		return
+	}
+
+	f.againstStrategy = getSetting(f, settings, ConfigBreakingStrategy, 0, againstFromString)
+	f.againstGitRef = getSetting(f, settings, ConfigBreakingGitRef, 1, func(s string) (string, bool) { return s, true })
```

Comment:
Might be worth commenting here the indices follow the ordering from the `ConfigurationParams.Items` from the request above.